### PR TITLE
Sovereign config flag and run type components

### PR DESCRIPTION
**Pull request by axenteoctavian**
- Sovereign config flag
- RunType Components implementation

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

File: cmd/proxy/main.go [hunk 1]
```diff
@@ -160,6 +163,11 @@ VERSION:
 		Name:  "start-swagger-ui",
 		Usage: "If set to true, will start a Swagger UI on the root",
 	}
+	// sovereignConfig defines a flag that specifies if what run type components should use
+	sovereignConfig = cli.BoolFlag{
```
Comment (mariusmihaic) [inline cmd/proxy/main.go:167]: Let's rename it, please. We do not use/change any config.
Should be simpler to just use `sovereign`

File: cmd/proxy/main.go [hunk 2]
```diff
@@ -383,6 +393,7 @@ func createVersionsRegistryTestOrProduction(
 		ctx.GlobalString(walletKeyPemFile.Name),
 		ctx.GlobalString(apiConfigDirectory.Name),
 		closableComponents,
+		ctx.GlobalBool(sovereignConfig.Name),
```
Comment (mariusmihaic) [inline cmd/proxy/main.go:396]: [optional]
We could move this flag before `closableComponents`

File: cmd/proxy/main.go [hunk 3]
```diff
@@ -561,6 +584,20 @@ func createVersionsRegistry(
 	return versionsFactory.CreateVersionsRegistry(facadeArgs, apiConfigParser)
 }
 
+func createManagedRunTypeComponents(factory runType.RunTypeComponentsCreator) (factory.RunTypeComponentsHandler, error) {
+	managedRunTypeComponents, err := runType.NewManagedRunTypeComponents(factory)
+	if err != nil {
+		return nil, err
+	}
+
+	err = managedRunTypeComponents.Create()
```
Comment (mariusmihaic) [inline cmd/proxy/main.go:593]: Simply return here

Comment (axenteoctavian) [reply cmd/proxy/main.go:593]: I can't. I need to return RunTypeComponentsHandler + error

File: factory/runType/runTypeComponents.go [hunk 4]
```diff
@@ -1,5 +1,11 @@
 package runType
 
+import (
+	"errors"
+)
+
+var errNilRunTypeComponents = errors.New("nil run type components")
```
Comment (mariusmihaic) [inline factory/runType/runTypeComponents.go:7]: We usually use another file `errors.go` in which we put this

**Review by mariusmihaic — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**